### ISSUE TITLE
Auto-complete non-exported global symbols from other modules

### ISSUE THREAD

**Issue by NotWearingPants** [microsoft/TypeScript#34539]
## Search Terms

import
export
unexported
symbols
auto import
completion
global
globals
intellisense

## Suggestion

tsserver can auto-complete exported symbols from other modules, and add an import statement that imports the symbol you selected.

It should also suggest **global symbols** from other modules that are ***not* exported**, and add the *export* keyword if the suggestion is selected.

## Use Cases

When I write a module I can't predict every use case of it, and I only export the global symbols that I think other modules would need to use. But when I'm working on a different module and I realize I want to use a symbol which is not exported by the other module (or maybe I don't even remember if it is or isn't exported), I would like my editor to export and import it for me instead of me having to find that symbol manually (because even go-to definition won't work) and export it, and then go back and auto-complete to auto-import.

## Examples

*module.ts*:
```ts
function foo(name = 'World') {
  console.log('Hello ' + name);
}

export function bar() {
  foo();
}
```

*app.ts*:
```ts
import { bar } from './module.ts';

bar();
```

I now want to customize the "hello" message, so I start typing `foo` and then wait for completion suggestions which include the function `foo` from `module.ts`.

After selecting that, the symbol `foo` is automatically exported and imported for me, and I can use it immediately:

*module.ts*:
```ts
export function foo(name = 'World') {
  console.log('Hello ' + name);
}

export function bar() {
  foo();
}
```

*app.ts*:
```ts
import { bar, foo } from './module.ts';

bar();
foo('TypeScript');
```

## Checklist

My suggestion meets these guidelines:

* [X] This wouldn't be a breaking change in existing TypeScript/JavaScript code
* [X] This wouldn't change the runtime behavior of existing JavaScript code
* [X] This could be implemented without emitting different JS based on the types of the expressions
* [X] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, etc.)
* [X] This feature would agree with the rest of [TypeScript's Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

**Comment by RyanCavanaugh** [comment]
For completions, this would be too much of a performance and predictability problem (tab-completing should really *never* change files other than the one you're editing).

For an error fix when you've typed the name and it doesn't resolve to anything, *if* one of the files you've already imported has an unexported declaration with that name, it seems entirely reasonable to offer to export it + import it into the current file.

Changing the title a bit to reflect that

**Comment by NotWearingPants** [comment]
I don't agree with your arbitrary rule, I think that it's a perfectly reasonable thing to offer in a tab-completion, as long as the consequences of choosing that option are reflected to the user (auto-imports are reflected in the tab-completion option, why not "auto-export+import"?).

But I guess a quick-fix is half of the solution, so it sounds good to me for now.

Just to clarify, my use case is converting a JS codebase with no module-system (not even common-js) to ES6-modules. With this quick-fix I can go through all of the undeclared symbol errors and apply the quick-fix and it will automatically add the import.
But you require that the file should already be imported, that means I'll have to add fake imports of all the files in every file in the codebase just to make the quick-fix work. It would be better to have it look at all of the files in the current project (`jsconfig.json`/`tsconfig.json`).
